src/components/widgets/MonthField.vue

```diff
@@ -0,0 +1,157 @@
+        { value: 7, label: 'August' },
+        { value: 8, label: 'September' },
+        { value: 9, label: 'October' },
+        { value: 10, label: 'November' },
+        { value: 11, label: 'December' }
+      ]
+    }
+  },
+
+  mounted() {
+    if (this.modelValue && this.modelValue.getMonth) {
+      this.selectedMonth = this.modelValue.getMonth()
+      this.selectedYear = this.modelValue.getFullYear()
+    }
+  },
+
+  computed: {
+    availableYears() {
+      const years = []
+      const startYear = this.minDate.getFullYear()
+      const endYear = this.maxDate.getFullYear()
+
+      for (let year = startYear; year <= endYear; year++) {
+        years.push(year)
+      }
+      return years.map(year => ({
+        value: year,
+        label: year
+      }))
+    },
+
+    availableMonths() {
+      const currentYear = this.selectedYear || this.availableYears[0].value
+      const startYear = this.minDate.getFullYear()
+      const endYear = this.maxDate.getFullYear()
+      const minMonth = this.minDate.getMonth()
+      const maxMonth = this.maxDate.getMonth()
+
+      return this.months
+        .filter(month => {
+          if (currentYear === startYear && currentYear === endYear) {
+            // If same year, only show months between start and end
+            return month.value >= minMonth && month.value <= maxMonth
+          } else if (currentYear === startYear) {
+            // If start year, only show months after start date
+            return month.value >= minMonth
+          } else if (currentYear === endYear) {
+            // If end year, only show months before end date
+            return month.value <= maxMonth
+          }
+          return true
+        })
+        .map(month => ({
+          value: month.value,
+          label: month.label
+        }))
+    }
+  },
+
+  watch: {
+    modelValue() {
+      console.log('modelValue', this.modelValue)
+      this.$options.silent = true
+      if (this.modelValue && this.modelValue.getMonth) {
+        this.selectedMonth = this.modelValue.getMonth()
+        this.selectedYear = this.modelValue.getFullYear()
+      }
+      this.$nextTick(() => {
+        this.$options.silent = false
+      })
+    },
+
+    selectedMonth(newMonth) {
+      if (this.$options.silent) return
+      const date = moment(new Date(this.selectedYear, newMonth, 1, 0, 0, 0, 0))
+      this.$emit('update:modelValue', date.startOf('month').toDate())
+    },
+
+    selectedYear(newYear) {
+      if (this.$options.silent) return
+      this.$nextTick(() => {
+        this.selectedMonth = this.availableMonths[0].value
+        const date = moment(
+          new Date(newYear, this.selectedMonth, 1, 0, 0, 0, 0)
+        )
+        this.$emit('update:modelValue', date.startOf('month').toDate())
+      })
+    }
+  }
+}
+</script>
+
+<style scoped>
+.date-selector {
+  display: flex;
+  gap: 1rem;
+}
+
+.month-select,
+.year-select {
+  padding: 0.5rem;
+  border: 1px solid #ddd;
```

Comment:
Maybe use a variable (e.g. `$light-grey-light`) instead of `#ddd`.

Reply:
The classes were not used, I removed everything.